signup/signup.js

```diff
@@ -1,92 +1,117 @@
-function pwError(e) {
-  let myPw = e.target.value
-  if(!myPw.trim()) {
-    errorPw.textContent = '비밀번호를 입력해 주세요.'
-    pwInput.classList.add('inputError')
-  } else if(!pwCheck(myPw)) {
-    errorPw.textContent = '비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.'
-    pwInput.classList.add('inputError')
-  } else{
-    errorPw.textContent=''
-    pwInput.classList.remove('inputError')
+function emailError(email) {
+  if(!email.trim()) {
+    putError({input: emailInput, errorMessage: errorEmail}, '이메일을 입력해 주세요.')
+    return false;
+  }
+  if(!emailValid(email)) {
+    putError({input: emailInput, errorMessage: errorEmail}, '올바른 이메일 주소가 아닙니다.')
+    return false;
+  }
+  if(!checkDuplication(email)) {
+    putError({input: emailInput, errorMessage: errorEmail}, '이미 사용 중인 이메일입니다.')
+    return false;
   }
+  removeError({input: emailInput, errorMessage: errorEmail})
+  return true
 }
-function pwErrorCheck(e) {
-  if(pwInput.value!==passwordCheck.value) {
-    errorPwCheck.textContent = '비밀번호가 일치하지 않아요.'
-    pwInput.classList.add('inputError')
-  } else{
-    errorPwCheck.textContent=''
-    pwInput.classList.remove('inputError')
+
+emailInput.addEventListener('focusout', (e) => {
+  emailError(e.target.value)
+})
+
+// password
+function pwError(password) {
+  if(!password.trim() || !passwordValid(password)) {
+    putError({input: pwInput, errorMessage: errorPw}, '비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.')
+    return false;
   }
+  removeError({input: pwInput, errorMessage: errorPw})
+  return true
 }
 
-// 올바른 이메일 주소가 아닙니다
-function emailCheck(e) {
-  const E_STYLE = /([\w-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/
-  if(e.target.value.trim() && !E_STYLE.test(e.target.value)) {
-    errorMessage.textContent = '올바른 이메일 주소가 아닙니다.'
-    emailInput.classList.add('inputError')
-  } else{
-    errorMessage.content=''
-    emailInput.classList.remove('inputError')
+pwInput.addEventListener('focusout', (e) => pwError(e.target.value))
+
+// check password
+function checkPwError(password) {
+  if(!password.trim() || !passwordValid(password)) {
+    putError({input: passwordCheck, errorMessage: errorPwCheck}, '비밀번호가 일치하지 않아요.')
+    return false;
   }
+  removeError({input: passwordCheck, errorMessage: errorPwCheck})
+  return true
 }
 
-// 로그인 했을 때 /folder로 이동 & 확인해주세요
-function login(e) {
-  if(emailInput.value!=="test@codeit.com" && pwInput.value!=="codeit101") {
-    return true
-  } else return false
+passwordCheck.addEventListener('focusout', (e) => checkPwError(e.target.value))
+
+//password toggle
+eye.addEventListener('click', () => togglePW(pwInput, eye))
+
+//check password toggle
+eyeCheck.addEventListener('click', () => togglePW(passwordCheck, eyeCheck))
+
+//로그인 했을 때 /folder로 이동 & 확인해주세요
+function signUpApi(mail, pw) {
+  const isEmailValid = emailError(mail)
+  const isPasswordValid = pwError(pw)
+  const isPasswordCheckValid = checkPwError(pw)
+  if(isEmailValid && isPasswordValid && isPasswordCheckValid) {
```

Comment:
되는 조건들에 한해서 실행하는 것도 좋지만 depth 가 깊어져서 이 부분은 미리 return 해주시는 걸 추천드려요!
```suggestion
  if(!isEmailValid || !isPasswordValid || !isPasswordCheckValid) return
```